scripts/cache_dataset.py

```diff
@@ -25,10 +25,15 @@ def main(args: RayCachedLMDatasetConfig):
     for split in ["train", "validation"]:
         # connect or start the actor
         batch_tokenizer = BatchTokenizer(tokenizer)
-        source = args.get_shard_source(split)
-
-        cache_dataset(os.path.join(args.cache_dir, split), source, batch_tokenizer)
-        print(f"Finished caching {split} to {args.cache_dir}/{split}.")
+        try:
+            source = args.get_shard_source(split)
+        except ValueError as e:
+            print(f"Skipping {split} because it doesn't exist: {e}")
```

Comment:
probably we should have has_split(x) on LMDatasetConfig

Reply:
Done